plugins/plugin-pullrequest-parent/che-plugin-pullrequest-server/src/main/java/org/eclipse/che/plugin/pullrequest/server/ContributionProjectTypeCleaner.java

```diff
@@ -0,0 +1,107 @@
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.api.core.notification.EventSubscriber;
+import org.eclipse.che.api.git.shared.event.GitRepositoryDeletedEvent;
+import org.eclipse.che.api.project.server.ProjectManager;
+import org.eclipse.che.api.project.server.impl.NewProjectConfigImpl;
+import org.eclipse.che.api.project.server.impl.RegisteredProject;
+import org.slf4j.Logger;
+
+/** @author Vitalii Parfonov */
+@Singleton
+public class ContributionProjectTypeCleaner implements EventSubscriber<GitRepositoryDeletedEvent> {
+
+  private static final Logger LOG = getLogger(ContributionProjectTypeCleaner.class);
+
+  private EventService eventService;
+  private Provider<ProjectManager> projectManagerProvider;
+
+  @Inject
+  public ContributionProjectTypeCleaner(
+      EventService eventService, Provider<ProjectManager> projectManagerProvider) {
+    this.eventService = eventService;
+    this.projectManagerProvider = projectManagerProvider;
+  }
+
+  @PostConstruct
+  private void subscribe() {
+    eventService.subscribe(this);
+  }
+
+  @PreDestroy
+  private void unsubscribe() {
+    eventService.unsubscribe(this);
+  }
+
+  private void onGitRepositoryDeletedEventReceived(String projectPath) {
+    try {
+      ProjectManager projectManager = projectManagerProvider.get();
+      projectManager.removeType(projectPath, CONTRIBUTION_PROJECT_TYPE_ID);
+      RegisteredProject project = projectManager.get(projectPath).get();
+      Map<String, List<String>> attributes = project.getAttributes();
+      attributes.remove(CONTRIBUTE_LOCAL_BRANCH_NAME);
+      attributes.remove(CONTRIBUTE_TO_BRANCH_VARIABLE_NAME);
+
+      NewProjectConfigImpl projectConfig =
+          new NewProjectConfigImpl(
+              project.getPath(),
+              project.getType(),
+              project.getMixins(),
+              project.getName(),
+              project.getDescription(),
+              attributes,
+              null,
+              project.getSource());
+
+      projectManager.update(projectConfig);
+    } catch (NotFoundException e) {
+      // throw if given project not found
+      LOG.error(
+          "Project {} not found for removing {} type",
+          projectPath,
+          CONTRIBUTION_PROJECT_TYPE_ID,
+          e);
+    } catch (ServerException e) {
+      e.printStackTrace();
```

Comment:
Probably `LOG.error` with simple explanation in message would be better.

Reply:
thx, i will fix